### PR TITLE
F/interactive contact section

### PR DESCRIPTION
**Pull request by whittlbc**
Migrating functionality of our marketing site's contact section (written in vanilla js) to react.
Added a reusable `FormInput` component with built in validation.

### CODE REVIEW

File: src/components/home/Contact.js [hunk 1]
```diff
@@ -1,39 +1,153 @@
...
+    return isValid;
+  }
+  
+  sendContactInfo(info, isMobile) {
+    if (isMobile) {
+      $(this.submitMobileContainer).addClass('loading');
+    }
+    
+    isMobile ? this.onMobileComplete() : this.onDesktopComplete();
+    // axios.post('/api/contact', info).then((resp) => {
```
Comment (atn34) [inline src/components/home/Contact.js]: This code would be enabled once we have an actual API right? Maybe explain that with a comment here.

File: src/components/home/Contact.js [hunk 2]
```diff
@@ -28,6 +30,7 @@ class Contact extends Component {
     this.setSectionRef = this.setSectionRef.bind(this);
     this.setSchoolFieldRef = this.setSchoolFieldRef.bind(this);
     this.setEmailFieldRef = this.setEmailFieldRef.bind(this);
+    this.formValid = this.formValid.bind(this);
```
Comment (atn34) [inline src/components/home/Contact.js]: Nit: This seems to be adding code in a commit with the message "adding comments"

File: src/components/home/Contact.js [hunk 3]
```diff
@@ -158,28 +158,28 @@ class Contact extends Component {
...
   }
   
-	render() {
-		return (
-		  <section id="contact" ref={this.setSectionRef}>
-				<div className="container-fluid">
-					<div className="leading-contact-text">Want Quokka on your campus? Let us know!</div>
-					<div className="row">
-						<div className="col-md-6 col-sm-12 col-xs-12 contact-field-container" role="school">
+  render() {
```
Comment (atn34) [inline src/components/home/Contact.js]: Nit: squash these fixes into the commit where they were introduced?

File: src/components/home/Contact.js [hunk 4]
```diff
@@ -38,98 +71,77 @@ class Contact extends Component {
...
+    this.schoolField.clearInput();
+    this.emailField.clearInput();
+
+    setTimeout(() => {
+      this.setState({
+        status: status.STATIC,
+        school: '',
+        email: ''
+      });
+    }, 1000);
```
Comment (atn34) [inline src/components/home/Contact.js]: Nit personally I felt 1 second to be a little short for the user to get feedback that clicking did anything. Can we make it more like 3 seconds? And also extract it to a descriptively-named constant please.

**Review by atn34 — COMMENTED** [review]
Nice! Looks good.

A couple nits:
1. I thought it was customary to write in the present tense, imperative style in commits. Like "Add changes" instead of "Adding" or "Added". No worries if you prefer a different convention, but we should probably at least be consistent.
2. `npm start` and `npm run sass` both are giving me warnings right now. It'd be nice to at least fix the ones introduced in this PR.
3. Tests would be nice to have, but I'm not sure how to write good tests for react yet.

**Comment by whittlbc** [comment]
Thanks! 

1. Interesting. Never knew that. We can switch to present-tense for sure. Should make for shorter commit messages as well.
2. Good idea will fix those.
3. Yeah I agree tests would be nice to have, but I don't think we can afford to add them from a time perspective just yet. Maybe after we get a demo out.

**Pull request merged**